### PR TITLE
Update documentation for generative constructor with optional positio…

### PR DESCRIPTION
**Pull request by danielerat**
Update documentation for generative constructor with optional positional parameters

- Updated the example for generative constructors to reflect the use of optional positional parameters with default values.
- Clarified the behavior of the constructor to indicate that `x` and `y` have default values of 2.0 when not passed explicitly.

### CODE REVIEW

File: src/content/language/constructors.md [hunk 1]
```diff
@@ -51,12 +51,12 @@ To instantiate a class, use a generative constructor.
 <?code-excerpt "point_alt.dart (idiomatic-constructor)" plaster="none"?>
```
Comment (MaryaBelanger) [inline src/content/language/constructors.md:51]: Hey @danielerat, thanks for your PR. Our code-excerpts correspond to Dart files in the `/examples` folder of this repo; this one corresponds to [this "docregion" in this file](https://github.com/dart-lang/site-www/blob/main/examples/misc/lib/language_tour/classes/point_alt.dart#L7). Could you update that to match your changes as well? And then run  `./dash_site refresh-excerpts` in your terminal (instructions in the site readme [here](https://github.com/dart-lang/site-www?tab=readme-ov-file#refresh-code-excerpts)). Thanks!

Comment (danielerat) [reply src/content/language/constructors.md:51]: Yes sure.

File: src/content/language/constructors.md [hunk 2]
```diff
@@ -51,12 +51,12 @@ To instantiate a class, use a generative constructor.
...
-  double x = 2.0;
-  double y = 2.0;
+  // Instance variables to hold the coordinates of the point.
+  double x;
+  double y;
 
-  // Generative constructor with initializing formal parameters:
-  Point(this.x, this.y);
+  // Generative constructor with optional positional parameters and default values.
+  Point([this.x = 2.0, this.y = 2.0]);
```
Comment (parlough) [inline src/content/language/constructors.md:59]: What do you think about just dropping the default values and making these required positional parameters?

Optional positional parameters and specifying their default values seem orthogonal to this section on "generative constructors".

**Review by parlough — APPROVED** [review]
Thanks for the PR @danielerat and for updating both sources! I think dropping the initializers here makes a lot of sense.

I have one question to consider, but looks good to me :)

File: src/content/language/constructors.md (hunk 2 above)
Comment (danielerat) [reply src/content/language/constructors.md:59]: Hey @parlough, I guess that could work too!

The idea behind keeping default values was to retain the intent of the original code while showcasing optional positional parameters. That said, making the parameters required positional would work just as well.
